torch/_functorch/aot_autograd.py

```diff
@@ -1283,11 +1287,27 @@ def aot_dispatch_base(flat_fn, flat_args: List[Tensor], aot_config: AOTConfig, *
     return compiled_fn
 
 
-def assert_functional_graph(fx_g: torch.fx.Graph):
+# Returns the number of detected copy_
+def assert_functional_graph(fx_g: torch.fx.Graph, *, allow_input_mutations: bool = False) -> int:
+    placeholders = set()
+    copy_count = 0
+    # NB: It would also be nice to verify that the mutations all happen at the
+    # end, but we also do some administrative views after mutations so this
+    # isn't actually true.  (TODO: Could this cause problems for Inductor?)
```

Comment:
Hmm, I vaguely remember this being a problem, but I thought we "fixed" it by syncing both the inputs and outputs before doing any copies during tracing. But I might be wrong (do you know of an instance where this happens?).

Reply:
yeah it failed on `pytest test/functorch/test_aotdispatch.py -k test_input_output_view_mutate_multiple`